feat(curriculum): add string method workshop (#66936)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-employee-profile-validator/69de4308e652aaeb5ccdb03e.md

---
id: 69de4308e652aaeb5ccdb03e
title: Step 1
challengeType: 20
dashedName: step-1
---

# --description--

> In this workshop, you will practice working with Python's string methods.

Reviewer: This file is the only one in the workshop saved with CRLF line endings; the other twenty use LF:

```
$ file *.md
69de4308e652aaeb5ccdb03e.md: ASCII text, with CRLF line terminators
...all twenty others: ASCII text
```

Worth converting before merge so the whole block is consistent.

Start by creating a variable named `email` and assign it the string `alice.johnson@company.com`.

# --hints--

> You should define a variable called `email`.

Reviewer: Every other step uses "create a variable named".

```suggestion
You should create a variable named `email`.
```

---
id: 69de4308e652aaeb5ccdb03e
title: Step 1
challengeType: 20
dashedName: step-1
---

# --description--

In this workshop, you will practice working with Python's string methods.

> Start by creating a variable named `email` and assign it an `empty string`.

Reviewer 2: ```suggestion
Start by creating a variable named `email` and assign it an empty string.
```

> Start by creating a variable named `email` and assign it an `empty string`.

Reviewer 3: ```suggestion
In this workshop, you will practice working with Python's string methods.

Start by creating a variable named `email` and assign it the string `'alice.johnson@company.com'`.
```

> Start by creating a variable named `email` and assign it the string `'alice.johnson@company.com'`.

Reviewer 2: In the Python curriculum any time we say to use a string it does not have quotes around it. All others labs and workshops do this so we should not change it here
```suggestion
Start by creating a variable named `email` and assign it the string `alice.johnson@company.com`.
```

# --hints--

You should define a variable called `email`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable("email")`)) })
```

> You should assign a empty string value to `email`.

Reviewer 2: ```suggestion
You should assign an empty string to your `email` variable.
```

> You should assign the value `'alice.johnson@company.com'` to `email`.

Reviewer 2: ```suggestion
You should assign the string `alice.johnson@company.com` to `email`. Remember to enclose the text between either single or double quotes.
```

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable("email")
not node.is_empty() and (
  node.is_equivalent("email = ''")

> )

Reviewer 2: ```suggestion
_Node(_code).find_variable("email").is_equivalent("email = ''")
```

> )

Reviewer 2: ```suggestion
_Node(_code).find_variable('email').is_equivalent("email = 'alice.johnson@company.com'")
```
